core/modules/replica/ChunkLocker.cc

```diff
@@ -0,0 +1,183 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/ChunkLocker.h"
+
+// System headers
+#include <stdexcept>
+#include <tuple>        // std::tie
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+///////////////////////////////////////
+//                Chunk              //
+///////////////////////////////////////
+
+bool Chunk::operator==(Chunk const& rhs) const {
+    return  std::tie(databaseFamily, number) ==
```

Comment:
what is the benefit of using `std::tie` here?

Reply:
I guess it makes it [easier to write good comparison operators](https://stackoverflow.com/questions/6218812/implementing-comparison-operators-via-tuple-and-tie-a-good-idea). Is there any question of thread safety here?

Reply:
This is C++11-ish way (pattern) of doing non-trivial comparisons :-). It replaces an older form (which is also less safer):

return (databaseFamily == rhs.databaseFamily) and (number == res.number);

Reply:
Technically speaking, std::tie construct a tuple of a variable number of (variadic) parameters. Benefits of this pattern become more clear when the number of parameters is higher. insider this:
std::tie(p1, p2, p3, p4) == std::time(rhs.p1, rhs.p2, rhs.p3, rhs.p4)

Clearly, one may make more mistakes if doing the comparison in the old way.